vault_program/src/delegate_token_account.rs

```diff
@@ -0,0 +1,94 @@
+use jito_account_traits::AccountDeserialize;
+use jito_jsm_core::loader::{
+    load_signer, load_token_2022_program, load_token_account, load_token_mint, load_token_program,
+};
+use jito_vault_core::vault::Vault;
+use solana_program::{
+    account_info::AccountInfo, entrypoint::ProgramResult, msg, program::invoke_signed,
+    program_error::ProgramError, pubkey::Pubkey,
+};
+
+/// Processes the delegate token account instruction: [`crate::VaultInstruction::DelegateTokenAccount`]
+///
+/// Admin might call the instruction when the vault is airdropped or transferred tokens
+pub fn process_delegate_token_account(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    amount: u64,
+) -> ProgramResult {
+    let [vault_info, admin, token_mint, token_account, delegate, token_program_info] = accounts
+    else {
+        return Err(ProgramError::NotEnoughAccountKeys);
+    };
+
+    Vault::load(program_id, vault_info, false)?;
+    load_signer(admin, false)?;
+    load_token_mint(token_mint)?;
+
+    let mut vault_data = vault_info.data.borrow_mut();
+    let vault = Vault::try_from_slice_unchecked_mut(&mut vault_data)?;
+
+    vault.check_admin(admin.key)?;
+    if vault.supported_mint.eq(token_mint.key) {
+        msg!("Invalid Token mint");
+        return Err(ProgramError::InvalidAccountData);
+    }
+
+    load_token_account(token_account, token_program_info)?;
```

Comment:
i don't see anything linking token_account to not be token_mint. 

think you can do something like this to rug the vault:
- pass in a fake mint
- pass in the token account with the supported mint

Reply:
you want to protect the supported mint ATA of the vault. 

so you need to have the linkage between token_mint -> token_account{mint: token_mint} and ensure that vault.supported_mint != token_mint

Reply:
> i don't see anything linking token_account to not be token_mint.

That's right. I modified the code that pass `token_mint.key` to `load_token_account`, then checking make sure `token_mint.key` is match to `token_account.mint`.